src/app/filter.pipe.ts

```diff
@@ -0,0 +1,18 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({
+  name: "filter",
+  pure: false
+})
+export class FilterPipe implements PipeTransform {
+  transform(value: any, filterString: string, propName: string): any {
+    if (value.length === 0 || filterString === "") return value;
```

Comment:
rewrite 
 (value.length === 0 || filterString === "") return value; 
to 
 (!value.length || !filterString ) return value;